packages/wonder-blocks-tabs/src/components/tabs.tsx

```diff
@@ -86,8 +96,90 @@ export const Tabs = React.forwardRef(function Tabs(
         onTabSelected,
         "aria-label": ariaLabel,
         "aria-labelledby": ariaLabelledby,
+        activationMode = "manual",
     } = props;
 
+    const focusedTabId = React.useRef(selectedTabId);
+
+    React.useEffect(() => {
+        focusedTabId.current = selectedTabId;
+    }, [selectedTabId]);
+
+    const selectTab = React.useCallback(
+        (tabId: string) => {
+            if (tabId !== selectedTabId) {
+                // Select the tab only if it's not already selected
+                onTabSelected(tabId);
+            }
+        },
+        [onTabSelected, selectedTabId],
+    );
+
+    const handleKeyInteraction = React.useCallback(
+        (tabId: string) => {
+            // Move focus to the tab
+            const tabElement = document.getElementById(getTabId(tabId));
+            tabElement?.focus();
+
+            switch (activationMode) {
+                case "manual": {
+                    // Only update which tab is focused since we aren't activating
+                    // the tab yet
+                    focusedTabId.current = tabId;
+                    break;
+                }
+                case "automatic": {
+                    // Activate the tab. When selectedTabId is updated, focus will
+                    // be moved to the selected tab
+                    selectTab(tabId);
+                    break;
+                }
+            }
+        },
+        [activationMode, selectTab, focusedTabId],
+    );
+
+    const handleKeyDown = React.useCallback(
+        (event: React.KeyboardEvent<HTMLButtonElement>) => {
+            const currentIndex = tabs.findIndex(
+                (tab) => tab.id === focusedTabId.current,
+            );
+
+            switch (event.key) {
+                case keys.left: {
```

Comment:
This could be a future enhancement, but do you think the tabs will ever stack vertically on small viewport sizes? Or will they always lay horizontally? I'm wondering: should the ArrowUp / ArrowDown keys do anything?

Reply:
Good question! I asked design about the responsive behaviour of tabs and we are having them be horizontally scrollable ([Slack thread](https://khanacademy.slack.com/archives/C07CA2YR0BZ/p1740082550098109?thread_ts=1740071221.168509&cid=C07CA2YR0BZ)).

The `ArrowUp` and `ArrowDown` keys don't do anything based on the [ARIA APG Tabs Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/):

> If the tab list is horizontal, it does not listen for Down Arrow or Up Arrow so those keys can provide their normal browser scrolling functions even when focus is inside the tab list.

I've seen some implementations where the DownArrow will move focus to the tabpanel though. I decided to stick with the ARIA APG pattern instead. Let me know though if you have any suggestions!